File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for x, y in zip(arr1, arr2):
        combine = x | y
        bin_secret = bin(combine)[2:].zfill(n)
        secert = ''
        
        for i in bin_secret:
            if i == '1':
                secert += '#'
            else:
                secert += ' '
        answer.append(secert)
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i = 0; i < n; i++) {
            String map = "";
            String binary = Integer.toBinaryString(arr1[i]|arr2[i]);
            
            while (binary.length() != n) {
                binary = "0" + binary;
            }
            
            for (char b: binary.toCharArray()) {
                if (b == '1') {
                    map += "#";
                } else {
                    map += " ";
                }
            }
            
            answer[i] = map;
        }
        
        return answer;
    }
}